packages/package.go

```diff
@@ -192,72 +209,127 @@ func (p *Package) addFile(c *config.Config, info fileInfo, cgo bool) error {
 		if info.isCgo {
 			return fmt.Errorf("%s: use of cgo in test not supported", info.path)
 		}
-		p.XTest.addFile(c, info)
+		pb.xtest.addFile(c, info)
 	case info.isTest:
 		if info.isCgo {
 			return fmt.Errorf("%s: use of cgo in test not supported", info.path)
 		}
-		p.Test.addFile(c, info)
+		pb.test.addFile(c, info)
 	case info.category == protoExt:
-		p.Proto.addFile(c, info)
+		pb.proto.addFile(c, info)
 	default:
-		p.Library.addFile(c, info)
+		pb.library.addFile(c, info)
 	}
 	if strings.HasSuffix(info.name, ".pb.go") {
-		p.Proto.HasPbGo = true
+		pb.proto.hasPbGo = true
 	}
 
 	return nil
 }
 
-func (t *GoTarget) addFile(c *config.Config, info fileInfo) {
-	if info.isCgo {
-		t.Cgo = true
+// isBuildable returns true if anything in the package is buildable.
+// This is true if the package has Go code that satisfies build constraints
+// on any platform or has proto files not in legacy mode.
+func (pb *packageBuilder) isBuildable(c *config.Config) bool {
+	return pb.firstGoFile() != "" ||
+		len(pb.proto.sources.strs) > 0 && c.ProtoMode == config.DefaultProtoMode
+}
+
+// firstGoFile returns the name of a .go file if the package contains at least
+// one .go file, or "" otherwise.
+func (pb *packageBuilder) firstGoFile() string {
+	goSrcs := []platformStringsBuilder{
+		pb.library.sources,
+		pb.binary.sources,
+		pb.test.sources,
+		pb.xtest.sources,
 	}
+	for _, sb := range goSrcs {
+		if sb.strs != nil {
+			for s, _ := range sb.strs {
+				if strings.HasSuffix(s, ".go") {
```

Comment:
Isn't a package buildable if it only has .s files?

Reply:
Nope. See [build.go:890](https://golang.org/src/go/build/build.go#L890).

I guess at least one .go file is needed to determine the package name. Not sure if there are other reasons.